packages/rum-core/test/common/observers/page-clicks.spec.js

```diff
@@ -139,4 +140,58 @@ describe('observePageClicks', () => {
 
     cancelHistorySub()
   })
+
+  describeIf(
+    'browsers supporting closest API',
+    () => {
+      let childElement = document.createElement('span')
+      ;[
+        {
+          name:
+            'should use customTransactionName when click target is a <button> descendant',
+          parentTagName: 'button',
+          customTransactionName: 'button-custom-name',
+          expected: 'Click - button-custom-name'
+        },
+        {
+          name:
+            'should use customTransactionName when click target is a <a> descendant',
+          parentTagName: 'a',
+          customTransactionName: 'a-custom-name-html',
+          expected: 'Click - a-custom-name-html'
+        },
+        {
+          name:
+            'should not use customTransactionName when click target is not descendant from <button> nor <a>',
+          parentTagName: 'div',
+          customTransactionName: 'div-custom-name-html',
+          expected: 'Click - span'
+        }
+      ].forEach(({ parentTagName, customTransactionName, expected }) => {
+        it('should respect the custom transaction name in a nested HTML structure', () => {
+          unobservePageClicks = observePageClicks(transactionService)
```

Comment:
shouldnt we call `unObserve` at the end?

Reply:
it's being called in line 51:

```
  afterEach(() => {
    document.body.removeChild(containerElement)
    // Remove the added listener since window object is shared among all tests
    unobservePageClicks()
  })
```